src/compiler/checker.ts

```diff
@@ -30413,11 +30413,15 @@ namespace ts {
             const parameterRange = hasRestParameter ? min
                 : min < max ? min + "-" + max
                 : min;
-            const error = hasRestParameter ? Diagnostics.Expected_at_least_0_arguments_but_got_1
-                : parameterRange === 1 && args.length === 0 && isPromiseResolveArityError(node) ? isInJSFile(node)
-                    ? Diagnostics.Expected_0_arguments_but_got_1_TypeScript_may_need_a_JSDoc_hint_that_the_call_to_new_Promise_produces_a_Promise_void
-                    : Diagnostics.Expected_0_arguments_but_got_1_Did_you_forget_to_include_void_in_your_type_argument_to_Promise
-                : Diagnostics.Expected_0_arguments_but_got_1;
+            const isVoidPromiseError = !hasRestParameter && parameterRange === 1 && args.length === 0 && isPromiseResolveArityError(node);
+            if (isVoidPromiseError && isInJSFile(node)) {
+                return getDiagnosticForCallNode(node, Diagnostics.Expected_1_argument_but_got_0_new_Promise_need_a_JSDoc_hint_to_produce_a_0_that_can_be_called_without_arguments, "resolve");

```

Comment:
Is there any convenient way to get the name of the function? I tried `symbolName(node.symbol)` without success

Reply:
`resolveName(node.expression, node.expression.escapedText, SymbolFlags.Value, undefined, undefined, false)?.name`

Reply:
It may also be reasonable just to call it `resolve` since that’s the parameter name in the library file.

Reply:
Ok, I will change it to a straightforward string